visualization/app/codeCharta/state/store/metricData/metricData.actions.ts

```diff
@@ -0,0 +1,29 @@
+import { CCAction, RecursivePartial, MetricData } from "../../../codeCharta.model"
+
+// Plop: Append default property import here
+import { defaultEdgeMetricData } from "./edgeMetricData/edgeMetricData.actions"
+import { defaultNodeMetricData } from "./nodeMetricData/nodeMetricData.actions"
+
+export enum MetricDataActions {
+	SET_METRIC_DATA = "SET_METRIC_DATA"
+}
+
+export interface SetMetricDataAction extends CCAction {
+	type: MetricDataActions.SET_METRIC_DATA
+	payload: RecursivePartial<MetricData>
+}
+
+export type MetricDataAction = SetMetricDataAction
+
+export function setMetricData(metricData: RecursivePartial<MetricData> = defaultMetricData): MetricDataAction {
```

Comment:
This is not really a setter. The name seems rather confusing to me.

Reply:
See my comment on ActionCreators